app/models/provider_interface/candidate_pool_filter.rb

```diff
@@ -1,143 +1,93 @@
 module ProviderInterface
   class CandidatePoolFilter
-    include FilterParamsHelper
-    include ActionView::Helpers::TagHelper
-    include Rails.application.routes.url_helpers
+    include ActiveModel::Model
+    include ActiveModel::Attributes
 
-    FILTERS = %w[original_location subject study_mode course_type visa_sponsorship].freeze
+    # filter attributes
+    attribute :location
+    attribute :subject
+    attribute :study_mode
+    attribute :course_type
+    attribute :visa_sponsorship
 
-    attr_reader :filter_params
+    attr_reader :filters, :current_provider_user, :remove_filter, :suggested_location
```

Comment:
Not a blocker, but I think it would be clearer (if I understand correctly), if we said `remove_all_filters` instead of `remove_filter`.